Feature/recipe overview

shayeedjamal opened the pull request.

Commit:
Made recipeDetail page which #3

Retrieves data from id passed as parameter

Commit:
Changed Recipeoverview to RecipeDetail Card displays all the details 

Not the most beatifull UI.  Also We have to delete all the recipes that are not complete. You have to click on a liked recipe to see the details. When you click on a recipe that does not has any ingredients or not a array of ingredients it will give an error recipe.ingredients.map is not a function.

Commit:
Merge branch 'develop' into feature/recipe-overview

Commit:
Added the reactflip library and also added steps to recipedetails #3

On create recipe page i made the steps store in a array.

Commit:
Fixed the eslint errors

Commit:
Merge branch 'develop' into feature/recipe-overview

negam00 reviewed (commented):
![image](https://user-images.githubusercontent.com/34008254/120908037-e839d480-c666-11eb-9633-bb64e98e6f8a.png)
:(

Ilias-Github commented:
> ![image](https://user-images.githubusercontent.com/34008254/120908037-e839d480-c666-11eb-9633-bb64e98e6f8a.png)
> :(

Did you "npm install"?

negam00 commented on src/components/NavBar/NavBar.jsx:
```diff
@@ -21,6 +21,7 @@ export default function NavBar() {
             <Nav.Link href="/">Home</Nav.Link>
             <Nav.Link href="add-recipe">Add Recipe</Nav.Link>
             <Nav.Link href="likes">Liked Recipes</Nav.Link>
+            <Nav.Link href="recipedetail">Recipe Details</Nav.Link>
```
Should this be in the navbar? As the user taps the recipe on either the liked or created recipes page.

shayeedjamal replied:
true im gonna remove it

negam00 reviewed (changes requested):
![image](https://user-images.githubusercontent.com/34008254/120918636-a3895a00-c6b5-11eb-9c57-dbbc963ffee4.png)
It's not centered, also the steps could be stacked above each other, but thats not that important. 
Also a small description on what the user can/needs to do to see the details would be great.

Commit:
Added a instruction on top of the likes and recipedetails page.

on create recipe page put a form label to tell the user they should seperate the input by comma

Commit:
Merge branch 'develop' into feature/recipe-overview

Commit:
deleted the route to the recipedetail page

Ilias-Github reviewed (dismissed):
* Add margins to the text at the top and the recipe card
* The description of a recipe is cut off. That is normal behavior for a card in the feed. But when looking at the details of the recipe, I expect to see the full description of a recipe
* CamelCase shouldn't be part of the UI
* The text should follow the same notation. (the front of the card shows: camelCase, CamelCase and normal text)
* Because the steps are now in an array, the user shouldn't be adding "step 1 ..., step2 ..., etc.)
* The instructional text needs improvement gramatically

Commit:
Changed the camelCase problem and also the instructions.

On the create recipe i added  the text to tell the user how they should add the steps. Also made the recipedetails margin like that of the feed. The full description is now being shown to the user.

Commit:
Merge branch 'develop' into feature/recipe-overview

negam00 commented on src/pages/RecipeOverview/RecipeDetail.jsx:
```diff
@@ -0,0 +1,99 @@
+                  </ListGroupItem>
+                  <ListGroupItem>
+                    <Card.Text id="recipe-detail-description">
+                      Dietpreference: {recipe.dietary}
```
This isn't a real word, should be a blank space in between

negam00 commented on src/pages/RecipeOverview/RecipeDetail.jsx:
```diff
@@ -0,0 +1,99 @@
+                  </ListGroupItem>
+                  <ListGroupItem>
+                    <Card.Text className="recipe-detail-description">
+                      Mealtype: {recipe.mealtype}
```
Type of meal would be more user friendly

negam00 reviewed (changes requested):
![image](https://user-images.githubusercontent.com/34008254/121011061-7a8cc600-c796-11eb-8465-1ca90dd14caf.png)
There is no margin at the bottom, this looks a little weird

Commit:
changed the name of mealtype and dietpreference also added margin

Commit:
fixed the css of the feed

Commit:
fixed the css on the feed

negam00 reviewed (dismissed):
unlike ilias' work, this is actually considerable.

stonnyu reviewed (dismissed):
Sorry for the late review, I found flipping the card to be very entertaining

Commit:
Merge branch 'develop' into feature/recipe-overview
